idb/dummy/dummy_factory.go

```diff
@@ -15,7 +16,7 @@ func (df dummyFactory) Name() string {
 }
 
 // Build is part of the IndexerFactory interface.
-func (df dummyFactory) Build(arg string, opts idb.IndexerDbOptions, log *log.Logger) (idb.IndexerDb, chan struct{}, error) {
+func (df dummyFactory) Build(config *pgxpool.Config, opts idb.IndexerDbOptions, log *log.Logger) (idb.IndexerDb, chan struct{}, error) {
```

Comment:
This interface should be DB agnostic, we shouldn't use `pgxpool` outside of the postgres implementation.